modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -455,7 +455,11 @@ void installTableColumnStyleClassListener() {
     NestedTableColumnHeader getNestedColumnHeader() { return nestedColumnHeader; }
     void setNestedColumnHeader(NestedTableColumnHeader nch) { nestedColumnHeader = nch; }
 
-    TableHeaderRow getTableHeaderRow() { return tableHeaderRow; }
+    /**
+     *
+     * @return The @link {@link TableHeaderRow} associated with this TableColumnHeader.
+     */
+    protected TableHeaderRow getTableHeaderRow() { return tableHeaderRow; }
```

Comment:
Not sure where nor how to put this ... but I think there are a couple of more methods that should be exposed/added:

expose:

    /**
     * Called by the parent header when this {@code TableColumnHeader} is removed
     * from its children. This method
     * must implement any logic necessary to clean up itself. 
     */
    protected void dispose() {

Reason: Implementations of custom column headers usually wire custom listeners (or other resources) that require to add custom cleanup. Currently, there's no way to do so. If done, it must be done also in NestedTableColumnHeader.

add: access to changeListenerHandler (could be implemented just the same way as in SkinBase without exposing the internal LambdaMultiplePropertyChangeListenerHandler)

reason: custom header typically add additional changeListeners - doing so by api that's already available would cleanup those wiring automatically in super's dispose. Currently, that can only be done by either re-implementing something like the lambdaHandler (*cough by c&p) or reflectively accessing the internal field (*cough again).